src/ptvsd/wrapper.py

```diff
@@ -2282,6 +2283,36 @@ def on_setBreakpoints(self, request, args):
 
         pydevd_request = copy.deepcopy(request)
         del pydevd_request['seq']  # A new seq should be created for pydevd.
+
+        # Validate breakpoints and adjust their positions.
+        args = pydevd_request['arguments']
+        if 'breakpoints' in args and 'path' in args['source']:
+            path = args['source']['path']
+            try:
+                with open(path) as f:
```

Comment:
For `attach` scenarios with path mappings this will not find the file.

Reply:
I think that this code should be in `_pydevd_bundle.pydevd_process_net_command_json._PyDevJsonCommandProcessor.on_setbreakpoints_request` (as @karthiknadig pointed, it requires internal `pydevd` apis to work properly).

Reply:
@fabioz I had talked about this with @int19h. Basically there are 2 parts to this change, validation and adjustment. I think validation part is well defined. The adjustment part is not yet clear about the possible set of settings we might need since it is subjective to user. So for now we wanted get something simple built and try it out. I definitely think that the validation part should be in pydevd. As for the adjustment part this may have to be done in ptvsd depending on how the settings will turnout for this. 

To keep it simple and avoid making the split when we are trying it out we decided to do it entirely here. Once we have some feedback from the internal loop we will move this into pydevd. At that point we can decide how we want to handle the settings.

Reply:
@karthiknadig, I have updated the PR to handle path mapping.